src/pages/Overview/index.js

```diff
@@ -1,16 +1,622 @@
+    // Stop propagation from going to the next page
+    event.stopPropagation();
+    const { dispatch } = this.props;
+    // dispatch an action to favorite controller
+    dispatch({
+      type: 'user/removeControllerFromFavorites',
+      payload: controller,
+    });
+  };
+
+  showDrowpdown = id => {
+    const dropdownElem = document.getElementById(id);
+    if (dropdownElem.style.display === 'none') {
+      dropdownElem.style.display = 'block';
+    } else {
+      dropdownElem.style.display = 'none';
+    }
+  };
+
+  saveRuns = row => {
+    const { totalResultData } = this.state;
+    const { dispatch } = this.props;
+    const objIndex = totalResultData.findIndex(x => x.key === row.key);
+    totalResultData[objIndex].saved = true;
+    this.setState(
+      {
+        totalResultData,
+      },
+      () => {
+        dispatch({
+          type: 'privatedatastore/updatePrivateControllers',
+          payload: totalResultData,
+        });
+        this.getSeperatedResults();
+      }
+    );
+  };
+
+  deleteResult = row => {
+    const { totalResultData } = this.state;
+    const { dispatch } = this.props;
+    const updatedResult = totalResultData.filter(x => x.key !== row.key);
+    this.setState(
+      {
+        totalResultData: updatedResult,
+      },
+      () => {
+        dispatch({
+          type: 'privatedatastore/updatePrivateControllers',
+          payload: totalResultData,
+        });
+        this.getSeperatedResults();
+      }
+    );
+  };
+
+  onToggleManageRunDropdown = isOpen => {
+    this.setState({
+      isOpen,
+    });
+  };
+
+  onSelectManageRunDropdown = () => {
+    const { isOpen } = this.state;
+    this.setState({
+      isOpen: !isOpen,
+    });
+  };
+
+  navigateToRunResult = (text, row) => {
+    const { dispatch } = this.props;
+    this.markResultSeen(row);
+    dispatch({
+      type: 'privatedatastore/updateSelectedPrivateController',
+      payload: text,
+    }).then(() => {
+      dispatch(
+        routerRedux.push({
+          pathname: 'private/runresults',
+        })
+      );
+    });
+  };
+
+  navigateToExpiringResult = () => {
+    const { dispatch } = this.props;
+    dispatch(
+      routerRedux.push({
+        pathname: 'private/expiringresults',
+      })
+    );
+  };
+
+  markResultSeen = controller => {
+    const { dispatch } = this.props;
+    dispatch({
+      type: 'user/markControllerSeen',
+      payload: controller,
+    });
+  };
+
+  removeControllerFromSeen = controller => {
+    const { dispatch } = this.props;
+    dispatch({
+      type: 'user/removeControllerFromSeen',
+      payload: controller,
+    });
+  };
+
   render() {
+    const { newData, unlabledData, isOpen } = this.state;
+    const { favoriteControllers, seenControllers } = this.props;
+
+    const newDataColumns = [
+      {
+        title: 'Result',
+        dataIndex: 'result',
+        key: 'result',
+        render: (text, row) => {
+          let isSeen = false;
```

Comment:
I'm getting the following error here:

TypeError: Cannot read property 'forEach' of undefined
render
src/pages/Overview/index.js:212
```
  209 | render: (text, row) => {
  210 |   let isSeen = false;
  211 |   if (seenControllers !== []) {
> 212 |     seenControllers.forEach(item => {
      | ^  213 |       if (item.key === row.key) {
  214 |         isSeen = true;
  215 |       }
```

Reply:
@gurbirkalsi could you please check the state in Redux DevTools? Or try opening accessing dashboard in incognito? I have changed the codebase now. Let me know if it works for you.